docker-compose.yml

```
@@ -213,6 +213,7 @@ services:
         - ALLOW_ADMIN_EMAIL_LOGIN=${ALLOW_ADMIN_EMAIL_LOGIN:-n}
         - IPV4_NETWORK=${IPV4_NETWORK:-172.22.1}
         - SOGO_EXPIRE_SESSION=${SOGO_EXPIRE_SESSION:-480}
+        - SOGO_URL_ENCRYPTION_KEY=${SOGO_URL_ENCRYPTION_KEY:-SOGoSuperSecret0}
```

IIRC the default `SOGoSuperSecret0` did not work in your tests? Should we maybe pick some other default value, or leave it blank and refuse to start if not appropriately set? Otherwise it might use unsecure keys without anyone noticing.

---

In my tests, I had a leading #, which resulted in 17 chars instead of the required 16 (128 bits).
It must be exactly 16 chars, otherwise it throws errors.

---

ACK. Makes sense. 

But honestly I'm still not a fan of silently using some hard-coded secret when `mailcow.conf` isn't updated for whatever reasons. Should we maybe add a `if [ "$SOGO_URL_ENCRYPTION_KEY" = "SOGoSuperSecret0"] && echo "Define encryption key. && exit 1` (pseudo-code) during SOGo-startup to make user aware to configure key properly?

---

If the Encryption Key is empty, SOGo will also default to `SOGoSuperSecret0`. That's why I also set it in docker-compose.yml
https://www.sogo.nu/files/docs/SOGoInstallationGuide.html#:~:text=auxiliary%20mail%20accounts.-,S,and%20the%20DAV%20url%20will%20change.%20Default%20value%20is%20SOGoSuperSecret0.,-D

---

I know. My point is just that we should never fallback to some insecure, hard-coded keys even if SOGo does. I wouldn't see this as best-practise.

Current implementation has the potential of insecure, default key to be used silently. We should error-out if `SOGO_URL_ENCRYPTION_KEY` not specified.